cpp/src/parquet/encoding-test.cc

```diff
@@ -314,6 +318,185 @@ TEST(TestDictionaryEncoding, CannotDictDecodeBoolean) {
   ASSERT_THROW(MakeDictDecoder<BooleanType>(nullptr), ParquetException);
 }
 
+// ----------------------------------------------------------------------
+// Shared arrow builder decode tests
+class TestDecodeArrow : public ::testing::Test {
+ public:
+  void SetUp() override {
+    InitDataInputs();
+    SetupEncoderDecoder();
+  }
+
+  void InitDataInputs() {
+    // Build a dictionary array and its dense representation for testing building both
+    auto dict_values = ::arrow::ArrayFromJSON(::arrow::binary(), "[\"foo\", \"bar\"]");
+    auto dtype = ::arrow::dictionary(::arrow::int8(), dict_values);
```

Comment:
This is probably because I haven't worked with the `arrow::DictionaryType` before, and I didn't immediately realize that `::arrow::dictonary` is a factory function which returns an `arrow::DictionaryType` value, but I found the name `dtype` to be a bit overly generic in this context. I think it would be nice to use a consistent naming convention (e.g. `dict_type`) to parallel `dict_values`. Similarly, II think it would be nice to rename `indices` to `dict_indices`.